src/components/App.js

```diff
@@ -665,17 +159,17 @@ class App extends Component {
 App.propTypes = {}
 
 const mapStateToProps = (reduxState) => {
-  return reduxState.chat;
+  return { ...reduxState.chat, ...reduxState.alert };
 };
 
 const mapDispatchToProps = (dispatch) => {
   return {
-    addNewMessage: ({ key, fromUsername, maybeSenderId, message }) =>
-      dispatch(addMessage({ key, fromUsername, maybeSenderId, message })),
-    addNewUserStatus: ({ fromUsername, userStatus, created }) =>
-      dispatch(addUserStatus({ fromUsername, userStatus, created })),
-    clearAllMessages: () => dispatch(clearMessages()),
-    setUsername: (username) => dispatch(setUsername(username))
+    sendMessage: ({ message, username }) =>
+    dispatch(sendMessage({ message, username })),
+    initConnection: (pincode) => dispatch(initConnection(pincode)),
+    initApplication: () => dispatch(initApplication()),
+    setUsername: (username) => dispatch(setUsername(username)),
+    dismissAlert: () => dispatch(dismissAlert())
   };
 };
 
```

Comment:
@Spetastian Awesome work man! `App.js` looks fantastically clean.

Reply:
@MartyTheeMartian You mentioned to me in another context that eliminating all state from the `App` component is desirable. Do you think so in this case?

Reply:
I dont think its a bad thing to have both Redux state and some internal component state. If all it does is to modify that specific components rendering but is not dependent on any external data or asynchronous operation or affects any sibling component then we could just use `this.setState(..)` and `this.state`.

But its more of a consideration from case to case. We dont want our components to be bloated.

@MartyTheeMartian What are your opinions about that?

Reply:
@Spetastian According to my understanding, @MartyTheeMartian wasn't saying that mixing some local state and redux state is bad in general, but that having any local state in `App` in particular means that any time any of that state changes, the entire app will re-render, because `App` _and all its subcomponents will re-render_, aka the entire application.

Reply:
Yes, that is a thing to consider. That is why i dont not want App.js to be the component that connects to the Redux State and propagates it to all its children. Instead it just composes some of the "page" or "area" components that each connects to Redux and resolves the part of the state used in that context.

Also we should use Reacts `PureComponent` more in order to prevent unnecessary re-renders.